js/index.js

```diff
@@ -1,42 +1,76 @@
 // ITERATION 1
 
 function updateSubtotal(product) {
-  console.log('Calculating subtotal, yey!');
-
-  //... your code goes here
+  //console.log('Calculating subtotal, yey!');
+  const price = product.querySelector('.price span');
+  const quantity = product.querySelector('.quantity input');
+  let calc = price.innerText * quantity.value;
+  const subTotal = product.querySelector('.subtotal span');
+  subTotal.innerText = calc;
+  return calc;
 }
 
 function calculateAll() {
   // code in the following two lines is added just for testing purposes.
   // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
+  //const singleProduct = document.querySelector('.product');
+  //updateSubtotal(singleProduct);
   // end of test
+  let sum = 0;
 
   // ITERATION 2
-  //... your code goes here
+  const cart = [...document.getElementsByClassName('product')];
+  cart.forEach((product) => (sum += updateSubtotal(product)));
```

Comment:
use reduce!!!

Reply:
you want a sum from an array -> reduce!!!
```js
const sum = cart.reduce((agg, product) => agg + updateSubtotal(product)), 0)
```

Reply:
also the name cart is misleading, since it's not a cart, but an array of products as Html Elements. 

you don't even need it, because you don't re-use cart so you could just write 
`[...document.getElementsByClassName('product')].reduce ...`